sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -546,113 +518,24 @@ public Registration(Class<T> proxyClass, List<PropertyDescriptor> beanInfo) {
    * @param iface The interface object to manually register.
    */
   public static synchronized void register(Class<? extends PipelineOptions> iface) {
-    checkNotNull(iface);
-    checkArgument(iface.isInterface(), "Only interface types are supported.");
-
-    if (REGISTERED_OPTIONS.contains(iface)) {
-      return;
-    }
-    validateWellFormed(iface, REGISTERED_OPTIONS);
-    REGISTERED_OPTIONS.add(iface);
+    CACHE.get().register(iface);
   }
 
   /**
    * Resets the set of interfaces registered with this factory to the default state.
+   * IMPORTANT: this is marked as experimental cause the safety usage of
```

Comment:
`IMPORTANT: this is marked as experimental cause the safety usage of this method is delegated to the caller.` -> `<p>IMPORTANT: this is marked as experimental because the correct usage of this method requires appropriate synchronization beyond the scope of this method.`